src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,134 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTORY = .75f;
+    private int size;
+    private int capacity;
+    private Node<K, V>[] dataTable;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        dataTable = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > capacity * LOAD_FACTORY) {
+            resize();
+        }
+        if (checkKey(key, value)) {
+            putNewNode(getHash(key), key, value, dataTable);
+            ++size;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = dataTable[getIndex(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getCapacity() {
+        return capacity;
+    }
+
+    @Override
+    public boolean containsKey(Object key) {
+        Node<K, V> node = dataTable[getIndex((K)key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return true;
+            }
+            node = node.next;
+        }
+        return false;
+    }
+
+    @Override
+    public Collection<V> values() {
+        Collection<V> collection = new ArrayList<>();
+        for (Node<K, V> node: dataTable) {
+            while (node != null) {
+                collection.add(node.value);
+                node = node.next;
+            }
+        }
+        return collection;
+    }
+
+    @Override
+    public void clear() {
+        dataTable = new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+    }
+
+    private boolean checkKey(K key, V value) {
```

Comment:
checkKey is not correct name for this method. You work not only with the key, but with value as well. And you not only check something - you change value in the table.
I sggest to delete this method and delete putNewNode() method. All logic you can have in put() - it will be about 20 lines - not to much.